configuration.rst

```diff
@@ -3,7 +3,14 @@
 Configuration
 =============
 
-PostgREST reads a configuration file to determine information about the database and how to serve client requests. There is no predefined location for this file, you must specify the file path as the one and only argument to the server:
+Without configuration, PostgREST won't be able to serve any requests. At the minimum it needs either a role to serve anonymous requests with - or a secret to use for JWT authentication. To connect to a database it will read `libpq environment variables <https://www.postgresql.org/docs/current/libpq-envars.html>`_ by default. The user specified in ``PGUSER`` is also known as the authenticator role. For more information about the anonymous vs authenticator roles see the :ref:`roles`.
```

Comment:
Hm, I think it's a bit confusing mentioning the LIBPQ env vars right away. Perhaps we can move the PGUSER, authenticator and pg docs to another section? Could be inside "Environment variables" as "LIBPQ environment variables".

Reply:
I think the libpq environment variables are different from the other environment variables. Especially because of how they interact with the db-uri string. I'm think about adding a separate section for "database connection" in general, which would describe the different ways to do so.

Reply:
Didn't go with a separate section, but still expanded on that. WDYT? Open for suggestions.

Reply:
> Hm, I think it's a bit confusing mentioning the LIBPQ env vars right away.

I thought about that part - and I feel like the most important thing on the configuration reference page is to get the facts / behaviour exactly right. I didn't touch the tutorials too much on purpose: Those still provide the connection string via plain config file, because that's easy enough to understand.

But especially the interaction between connection string and libpq environment variables deserves notice. Even when using a dedicated connection string, libpq will still read those parameters that are not set from env variables. So that could create some confusion.

Reply:
>  WDYT? Open for suggestions.

I like it, looks much better now.

> configuration reference page is to get the facts / behaviour exactly right. I didn't touch the tutorials too much on purpose

Agree, the reference doesn't have to include a quickstart, that's better left for the tutorials.